website_sale_preset_carts/models/time_frame.py

```diff
@@ -113,20 +121,88 @@ def adapt_qty_for_household(line):
             self.env['sale.order.line'].create(line)
         return sale_order
 
+    def compute_cart_amount(self, subscriber, lines):
+        amount = [line['price_unit'] * line['product_uom_qty'] * subscriber.nb_household  for line in lines][0]
+        return amount
+
     @api.model
     def generate_sale_orders(self):
+        blocked_order_obj = self.env['time.frame.blocked.order']
+
         for frame in self:
 
             unit_lines = frame._prepare_order_lines()
 
+            # we first get the customers with the corresponding subscription
             customers = (
                 self.env['res.partner']
-                    .search([('subscription_id', '=', frame.subscription_id.id),
-                             '|', ('cart_suspended_date', '=', False),
-                                  ('cart_suspended_date', '<=', frame.delivery_date)])
+                    .search([('subscription_id', '=', frame.subscription_id.id)])
             )
 
-            for customer in customers:
-                order = self._create_sale_order(customer, unit_lines)
-                frame.sale_orders += order
-                customer.write({'last_website_so_id': order.id})
+            # we then filter based on suspended date if suspended
+            subscribers = customers.filtered(
+                lambda cust: not cust.suspend_cart or (
+                        cust.cart_suspended_from > frame.delivery_date 
+                        or cust.cart_suspended_date < frame.delivery_date
+                    )
+                )
```

Comment:
Can't you just call `is_subscribed(self, date=None)`?

Reply:
no has is_subscribed is not a field. maybe is_subscribed should be set as a stored field. then I think we could rely on it for the filtering. but we don't have enough budget to figure that out and to do it.

I put a task for enhancement